src/client/connection.rs

```diff
@@ -44,14 +33,14 @@ use winauth::{windows::NtlmSspiBuilder, NextBytes};
 ///
 /// [`Client`]: struct.Encode.html
 /// [`Packet`]: ../protocol/codec/struct.Packet.html
-pub(crate) struct Connection {
-    transport: Framed<MaybeTlsStream, PacketCodec>,
+pub(crate) struct Connection<S: futures::AsyncRead + futures::AsyncWrite + Unpin + Send> {
+    transport: Framed<MaybeTlsStream<S>, PacketCodec>,
     flushed: bool,
     context: Context,
     buf: BytesMut,
 }
 
-impl Debug for Connection {
+impl<S: futures::AsyncRead + futures::AsyncWrite + Unpin + Send> Debug for Connection<S> {
```

Comment:
You might want to use futures::{AsyncRead, AsyncWrite} to save valuable horizontal space.